src/engine/SCons/Node/__init__.py

```diff
@@ -784,6 +785,25 @@ def built(self):
         for parent in self.waiting_parents:
             parent.implicit = None
 
+            # Handle issue where builder emits more than one target and
```

Comment:
This looks fine to me. I guess my only question would be why aren't the other targets already in waiting_parents so they get cleared above?

Reply:
It's a combination of the first target hit by the taskmaster walk and the fact that the N targets aren't linked.  I'm going to continue looking at this, but this fix resolves a long standing issue so accepting the good enough, and then will continue working on the maybe better solution.